### PR TITLE
Replace lazy file

### PR DESCRIPTION
**Pull request by TheNeikos**
This PR supersedes #514 and closes it.

### CODE REVIEW

File: libimagstore/src/store.rs [hunk 1]
```diff
@@ -1,12 +1,12 @@
 use std::collections::HashMap;
-use std::fs::{File, remove_file};
+use std::fs::remove_file;
```
Comment (TheNeikos) [inline libimagstore/src/store.rs]: This here needs also to be removed

**Comment by TheNeikos** [comment]
This can be reviewed now

**Comment by matthiasbeyer** [comment]
Looks good to me.

If you tell me where to add tests and maybe some suggestions on what to test, I will provide you some commits containing tests, if you want me to...

File: libimagstore/src/file_abstraction.rs [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+                    map.insert(f.clone(), Cursor::new(vec));
+                    return Ok(());
+                },
+            };
+        }
+
+        pub fn remove_file(_: &PathBuf) -> Result<(), SE> {
+            Ok(())
+        }
+        pub fn copy(from: &PathBuf, to: &PathBuf) -> Result<(), SE> {
```
Comment (matthiasbeyer) [inline libimagstore/src/file_abstraction.rs]: Could you add a blank line above this one?

**Comment by matthiasbeyer** [comment]
@TheNeikos ping.

**Comment by matthiasbeyer** [comment]
@TheNeikos Please rebase on latest master,... I cannot merge. And you should fixup the annotation if possible! :smile:

**Comment by matthiasbeyer** [comment]
As soon as #652 is merged you can rebase and CI should succeed then.

**Comment by matthiasbeyer** [comment]
@TheNeikos please rebase onto latest master!

**Comment by matthiasbeyer** [comment]
Finally!

:tada: @TheNeikos :tada:

**Pull request merged**